image-scanner.sh

```diff
@@ -0,0 +1,80 @@
+#!/bin/bash
+
+# Container scanning is the process of analyzing components within containers to uncover 
+# potential security threats. It is integral to ensuring that the software remains secure 
+# as it progresses through the application life cycle. Container scanning takes its cues 
+# from practices like vulnerability scanning and penetration testing.
+
+# This script uses Trivy, a simple and comprehensive vulnerability scanner
+# for containers and other artifacts. Trivy can find vulnerabilities, 
+# IaC misconfigurations, secrets, SBOM discovery, Cloud scanning, Kubernetes security risks, 
+# and more. More information about Trivy can be found at https://trivy.dev/.
+
+# This shell script automates the process of scanning container images from a specific Docker 
+# organization (in this case, bahmni) using the Trivy vulnerability scanner. The script 
+# retrieves a list of Docker repositories from Docker Hub and runs Trivy scans on each image 
+# to generate security reports. These reports are saved as HTML files in a designated directory.
+
+# Get Docker Organization Name
+ORG="bahmni"
+
+# Check if trivy is installed and install it if not
+if ! command -v trivy &> /dev/null; then
+    echo "Trivy is not installed. Installing..."
+    wget https://github.com/aquasecurity/trivy/releases/download/v0.51.1/trivy_0.51.1_Linux-64bit.deb
+    sudo dpkg -i trivy_0.51.1_Linux-64bit.deb
+else
+    echo "Found trivy, using trivy v$(trivy -v | cut -d ' ' -f 2)"
+fi
+
+echo "Retrieving repository list ..."
+REPO_LIST=$(curl -s "https://hub.docker.com/v2/repositories/${ORG}/?page_size=100" | jq -r '.results|.[]|.name')
+
+# Get today's date in the desired format for folder naming
+TODAY_DATE=$(date +'%d-%m-%Y')
+# Define the root and sub directory name
+ROOT_DIR="container-scanner-reports"
+DIR="bahmni-${TODAY_DATE}"
+
+# Check if the root directory exists and create it if not
+if [ ! -d "$ROOT_DIR" ]; then
+    echo "Creating directory $ROOT_DIR"
+    mkdir "$ROOT_DIR"
+else
+    echo "Root Directory $ROOT_DIR already exists, proceeding"
+fi
+
+# Check if the root directory exists and create it if not
+if [ ! -d "$ROOT_DIR/$DIR" ]; then
+    echo "Creating directory $ROOT_DIR/$DIR"
+    mkdir "$ROOT_DIR/$DIR"
+else
+    echo "Directory $ROOT_DIR/$DIR already exists, proceeding"
+fi
+
+echo "Generating scan report...."
+
+# Iterate through each image and run Trivy scan
+for image in $REPO_LIST; do
+    # Extract image name and tag
+    image_name=$(echo "$image" | cut -d ':' -f 1)
+    tag="$(echo "$image" | cut -d ':' -f 2)"
```

Comment:
From the response from dockerhub, i dont see we are getting the image tag
```{
			"name": "openmrs",
			"namespace": "bahmni",
			"repository_type": "image",
			"status": 1,
			"status_description": "active",
			"description": "",
			"is_private": false,
			"star_count": 1,
			"pull_count": 20143,
			"last_updated": "2024-05-08T11:50:39.204925Z",
			"date_registered": "2021-10-25T06:07:48.264052Z",
			"affiliation": "",
			"media_types": [
				"application/vnd.docker.container.image.v1+json",
				"application/vnd.docker.distribution.manifest.list.v2+json",
				"application/vnd.oci.image.index.v1+json"
			],
			"content_types": ["image"],
			"categories": []
		},
```

@rahu1ramesh Please check once and verify if this extraction of image and tag is needed